website/docs/developer/research/research-bamboo.md

```diff
@@ -0,0 +1,164 @@
+  - admin
+```
+- Docker- It is also possible to run a job inside a docker image rather than the bamboo agent if required. Check this [documentation](https://hub.docker.com/r/atlassian/bamboo-agent-base) for more information.
+
+```yaml
+test job name:
+  key: TJN
+  docker:
+    image: alpine
+```
+- Artifact- Artifacts are files created by a job build that can be used in other stages. An artifact definition can be created to specify which artifacts to keep and share from a build job. In the example below, an artifact definition named `test-artifact` is created under the job `test job` which is shared with other builds and deployments. Only the file matching the pattern .xml will be stored under artifact directory. 
+
+```yaml
+test job:
+  artifacts:
+  - name: test-artifact
+    location: artifact
+    pattern: '*.xml'
+    shared: true
+```
+Check artifact [documentation](https://confluence.atlassian.com/bamboo/configuring-a-job-s-build-artifacts-289277071.html) for more information.
+
+- Checkout- This is a task to check out a repository for use by just one job. In the example below, the job named `test job` has a checkout task that will checkout to the default repo defined in the plan before the tasks executes.
+
+```yaml
+test job:
+  tasks:
+  - checkout:
+      description: Checkout Default Repository
+```
+Check this [documentation](https://confluence.atlassian.com/bamboo0800/checking-out-code-1077778795.html) for more information.
+
+
+## Running Bamboo specs
+In order to run the YAML definition file, it has to be stored at `bamboo-specs/bamboo.yml` or `bamboo-specs/bamboo.yaml`under the repository specified. Check this [documentation](https://confluence.atlassian.com/bamboo/bamboo-yaml-specs-938844479.html) for more information.
+
+:::important
+Make sure the linked repository have permissions to create plans within given project in order to execute the YAML definition and create a plan. 
+:::
+
+In order to run this YAML definition from github repo, user has to setup `Repository-stored Specs` on Bamboo CI/CD [website](https://ci.earthdata.nasa.gov/build/admin/create/newSpecs.action). From the Bamboo dashboard, Choose `Specs`->`Set up Specs repository` and then select `Build project` as `ORCA` and `Repository host` as `ORCA repo`.
+The next step is to create a `Webhook` on the source repository by going to the settings option so that Bamboo knows about new commits. Webhooks allow Github repositories to communicate with Bamboo. A webhook has been created for ORCA project `https://ci.earthdata.nasa.gov/rest/api/latest/repository/scan?repositoryId=375849419` which needs to be copied to the `webhook` section in cumulus-orca repo. Without adding webhook, Bamboo was not able to recognize the new Spec file added to the `cumulus-orca` repo. Email `nasa-data@lists.arc.nasa.gov` to NASA Github admin team to make that change. More instructions can be found [here](https://github.com/nasa/instructions/blob/master/docs/INSTRUCTIONS.md#org-owners)
```

Comment:
I am wondering from a security perspective if we should include our actual webhook link `https://ci.earthdata.nasa.gov/rest/api/latest/repository/scan?repositoryId=375849419` note that this page will be out there for public consumption. Is there a way to lookup the webhook after the fact? Might be good to remove the actual webhook URL and just say that we have one and this is where you can find it in Bamboo. Thoughts? @AndrewDorn  and @rizbi-hassan

Reply:
I think we should remove the actual URL. I can point to our plan in Bamboo. We can get the webhook from there